src/ol/style/IconImage.js

```diff
@@ -152,18 +159,22 @@ class IconImage extends EventTarget {
    * @return {HTMLImageElement|HTMLCanvasElement} Image element.
    */
   getHitDetectionImage(pixelRatio) {
-    if (!this.hitDetectionImage_) {
-      if (this.isTainted_()) {
+    if (!this.hitDetectionImage_[pixelRatio]) {
+      if (this.color_ && this.isTainted_()) {
         const width = this.size_[0];
         const height = this.size_[1];
-        const context = createCanvasContext2D(width, height);
+        const context = createCanvasContext2D(
+          width * pixelRatio,
+          height * pixelRatio
+        );
+        context.scale(pixelRatio, pixelRatio);
         context.fillRect(0, 0, width, height);
-        this.hitDetectionImage_ = context.canvas;
+        this.hitDetectionImage_[pixelRatio] = context.canvas;
       } else {
-        this.hitDetectionImage_ = this.image_;
+        this.hitDetectionImage_[pixelRatio] = this.image_;
```

Comment:
The case when it is tainted but no color is set needs to also fill the canvas, but it does not need to be scaled.

Reply:
I don't understand why we need it... no problem to do this change :-)

Reply:
I probably just understand, it's to be able to do the get the color on a pixel :-)

Reply:
Try loading an image from an external server, then the canvas is 'tainted' and this line will throw an error.
https://github.com/openlayers/openlayers/blob/2dd9342a982e14c068cdd8c6492d2c6f79e8f81f/src/ol/style/IconImage.js#L95
This is a browser security measure.

The fallback is to fill the entire icon with one color for hit detection.

Reply:
+1, Tested and working :-)